drivers/bus.py

```diff
@@ -3,17 +3,23 @@
 """
 
 
-class Bus:
-    def __init__(self, logger):
+class BusHandler:
+    def __init__(self, logger, input_queue=None, out={}):
         self.logger = logger
+        self.queue = input_queue
+        self.out = out
 
-    def publish(self, stream_id, data):
+    def publish(self, channel, data):
         with self.logger.lock:
-            dt = self.logger.write(stream_id, data)
-            pass
+            stream_id = channel  # TODO local maping of indexes
+            timestamp = self.logger.write(stream_id, data)
+            if channel in self.out:
```

Comment:
It the code tries to publish on undefined/unallocated channel, it should fail IMO.